cli/src/memo.rs

```diff
@@ -0,0 +1,22 @@
+use solana_sdk::instruction::Instruction;
+use solana_sdk::pubkey::Pubkey;
+use std::str::FromStr;
+
+pub trait WithMemo {
```

Comment:
Nice approach!

Reply:
@CriesofCarrots wdyt about this trait living in `spl-memo` instead of here?

Reply:
That actually makes a lot of sense. @bji are you willing to pr your trait here? https://github.com/solana-labs/solana-program-library/blob/master/memo/program/src/lib.rs

Reply:
...nevermind, as per discussion on Discord.